### ISSUE TITLE
ComposeUIViewController is broken since 1.6.0-beta01 - iOS 16

### ISSUE THREAD

**Issue by alaegin** [JetBrains/compose-multiplatform#4310]
**Describe the bug**
1.6.0-beta01 broke ComposeUIViewController usage on iOS 16.4. 
The app crashes when the view using ComposeUIViewController is removed from the screen hierarchy.

**Affected platforms**
- iOS

**Versions**
- Kotlin version*:  1.9.22
- Compose Multiplatform version*: 1.6.0-beta01, 1.6.0-beta02, 1.6.0-rc01 
- OS version(s)* (required for Desktop and iOS issues):  iOS 16.4.0, 16.6.0
- OS architecture (x86 or arm64): arm64?

**To Reproduce**
1. Go to [this reproduction sample](https://github.com/alaegin/compose-multiplatform/pull/1) (I've updated chat sample a little bit, just updated the compose version and changed the navigation logic.
2. Launch it on iOS 16.4 simulator.
3. Click two times on 'Toggle' button in the center of the screen. (First click switches from SwiftUI screen, the second one switches back and there is the place where crash occurs)
5. See error.

The usual stack trace of this issue is:

```
Uncaught Kotlin exception: kotlin.IllegalStateException: ComposeUIViewController.view should be attached to window
    at 0   Chat                                0x1050d3c73        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 
    at 1   Chat                                0x1050cd40f        kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 
    at 2   Chat                                0x1050cd62f        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 115 
    at 3   Chat                                0x1050cdaa3        kfun:kotlin.IllegalStateException#<init>(kotlin.String?){} + 115 
    at 4   Chat                                0x105771a27        kfun:androidx.compose.ui.window.ComposeContainer.<get-windowContainer>#internal + 567 
    at 5   Chat                                0x105773aff        kfun:androidx.compose.ui.window.ComposeContainer#objc:viewWillLayoutSubviews + 739 
    at 6   Chat                                0x1057799e7        _6f72672e6a6574627261696e732e636f6d706f73652e75693a75692f6f70742f6275696c644167656e742f776f726b2f383339343731666333323931333563372f636f6d706f73652f75692f75692f7372632f75696b69744d61696e2f6b6f746c696e2f616e64726f6964782f636f6d706f73652f75692f77696e646f772f436f6d706f7365436f6e7461696e65722e75696b69742e6b74_knbridge130 + 195 
    at 7   UIKitCore                           0x12160e867        -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 939 
    at 8   QuartzCore                          0x109ce8527        _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 439 
    at 9   QuartzCore                          0x109cf3287        _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 127 
    at 10  QuartzCore                          0x109c1d12f        _ZN2CA7Context18commit_transactionEPNS_11TransactionEdPd + 439 
    at 11  QuartzCore                          0x109c480f3        _ZN2CA11Transaction6commitEv + 635 
    at 12  QuartzCore                          0x109c49517        _ZN2CA11Transaction25flush_as_runloop_observerEb + 67 
    at 13  CoreFoundation                      0x1087d0c0f        __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 31 
    at 14  CoreFoundation                      0x1087cb57b        __CFRunLoopDoObservers + 511 
    at 15  CoreFoundation                      0x1087cba1f        __CFRunLoopRun + 947 
    at 16  CoreFoundation                      0x1087cb253        CFRunLoopRunSpecific + 583 
    at 17  GraphicsServices                    0x10f82bc9b        GSEventRunModal + 159 
    at 18  UIKitCore                           0x12116afef        -[UIApplication _run] + 867 
    at 19  UIKitCore                           0x12116ef3b        UIApplicationMain + 123 
    at 20  SwiftUI                             0x110fd334b        OUTLINED_FUNCTION_54 + 499 
    at 21  SwiftUI                             0x110fd31eb        OUTLINED_FUNCTION_54 + 147 
    at 22  SwiftUI                             0x1107cb473        OUTLINED_FUNCTION_16 + 87 
    at 23  Chat                                0x10491713b        $s4Chat6iOSAppV5$mainyyFZ + 39 
    at 24  Chat                                0x1049171eb        main + 11 (/Users/alexey/StudioProjects/tmp/compose-multiplatform/examples/chat/iosApp/iosApp/iOSApp.swift:<unknown>)
    at 25  dyld                                0x1074d5513        0x0 + 4417475859 
    at 26  ???                                 0x10763df27        0x0 + 4418952999 
    at 27  ???                                 0xf811ffffffffffff 0x0 + -571394202722631681 
```

Statements:
This behaviour doesn't work fine on iOS 16 since 1.6.0-beta01.
This behaviour works fine on iOS 17+ on any 1.6.0 version
This behaviour works fine on iOS 16 using 1.6.0-alpha01 and older.

**Expected behavior**
I expect everything to work without exceptions like on version 1.6.0-alpha01.

**Screenshots**

**Additional context**
[Reproduction link](https://github.com/alaegin/compose-multiplatform/pull/1)

**Comment by MatkovIvan** [comment]
@alaegin Thanks for the report.

Before the fix, you can disable a new behavior for `Popup`/`Dialog` (that requires that window reference to put these elements into it) via ​​`ComposeUIViewControllerConfiguration.platformLayers` flag

**Comment by MatkovIvan** [comment]
jfyi the fix was landed too late for 1.6.0-rc02, we'll prepare rc03 for that

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4310">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
